src/plugins/dam/components/batch-edit-modal/delegate.js

```diff
@@ -0,0 +1,91 @@
+import { reset } from 'redux-form';
+import noop from 'lodash/noop';
+
+import batchEditPatchAssets from '@triniti/cms/plugins/dam/actions/batchEditPatchAssets';
+import FormEvent from '@triniti/app/events/FormEvent';
+
+import {
+  SUFFIX_VALIDATE_FORM,
+  SUFFIX_WARN_FORM,
+} from '@triniti/app/constants';
+
+import schemas from './schemas';
+import { formNames } from '../../constants';
+
+class Delegate {
+  constructor(config) {
+    this.config = config;
+
+    /** @type {Pbjx} pbjx */
+    this.pbjx = config.pbjx;
+
+    /** @type {Function} */
+    this.dispatch = noop;
+
+    this.handleReset = this.handleReset.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+    this.handleValidate = this.handleValidate.bind(this);
+    this.handleWarn = this.handleWarn.bind(this);
+  }
+
+  getFormName() {
+    return formNames.BATCH_EDIT;
+  }
+
+  bindToComponent(component) {
+    this.component = component;
+    this.dispatch = component.props.dispatch;
+  }
+
+  createFormEvent(data, formProps) {
+    const command = schemas.updateNode.createMessage({
+      expected_etag: null,
+      node_ref: this.component.props.nodeRef,
+    });
+
+    return new FormEvent(command, formProps.form, data, formProps);
+  }
+
+  getInitialValues() {
+    // can be filled in later with the common values between each node
+    return {};
+  }
+
+  handleValidate(values, formProps) {
+    const formEvent = this.createFormEvent(values, formProps);
+    this.pbjx.trigger(formEvent.getMessage(), SUFFIX_VALIDATE_FORM, formEvent);
```

Comment:
@rsumilang i dont think any subscribers are actually picking this up. if you paste a mega long string into the title field, you dont get any warnings and can still press the button but that then logs an error in the guard